algorithms/src/merkle_tree/merkle_path.rs

```diff
@@ -82,9 +82,9 @@ impl<P: MerkleParameters> MerklePath<P> {
     /// Returns: (left, right)
     fn select_left_right_bytes(
         index: u64,
-        computed_hash: &<P::H as CRH>::Output,
-        sibling_hash: &<P::H as CRH>::Output,
-    ) -> Result<(<P::H as CRH>::Output, <P::H as CRH>::Output), MerkleError> {
+        computed_hash: &<P::LeafCRH as CRH>::Output,
+        sibling_hash: &<P::LeafCRH as CRH>::Output,
+    ) -> Result<(<P::LeafCRH as CRH>::Output, <P::LeafCRH as CRH>::Output), MerkleError> {
```

Comment:
Just to sanity check, shouldn't this be using a `TwoToOneCRH`? (I think only the first instance should be using the `LeafCRH`)